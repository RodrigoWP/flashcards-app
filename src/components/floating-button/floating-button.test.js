import React from 'react'
import { shallow } from 'enzyme'
import { FloatingButton } from '.'

it('Should be render correctly', () => {
    const Component = shallow(
        <FloatingButton onPress={jest.fn} />
    )

    expect(Component).toMatchSnapshot()
})

it('Should be call onPress when button is clicked', () => {
    const onPress = jest.fn()

    const component = shallow(
        <FloatingButton onPress={onPress} />
    )

    const render = component.find('TouchableHighlight')
    render.simulate('press')

    expect(onPress).toHaveBeenCalledTimes(1)
})
