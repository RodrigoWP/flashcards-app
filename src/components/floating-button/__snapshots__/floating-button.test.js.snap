// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be render correctly 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6197c6",
      "borderRadius": 70,
      "bottom": 10,
      "height": 70,
      "justifyContent": "center",
      "position": "absolute",
      "right": 10,
      "width": 70,
    }
  }
  underlayColor="#afcef3"
>
  <Icon
    allowFontScaling={false}
    color="#ffffff"
    name="add"
    size={32}
  />
</TouchableHighlight>
`;
